ansible/influxdb-meta/tasks/service.yml

```diff
@@ -0,0 +1,38 @@
+- name: Start the influxdb-meta service
+  service:
+    name: influxdb-meta
+    state: restarted
+    enabled: yes
+  register: influxdb_meta_started
+  when: influxdb_meta_start_service
+
+- name: Wait for influxdb-meta to come up
+  wait_for:
+    host: "{{ influxdb_meta_http_bind_address_hostname | default('localhost', true) }}"
+    port: "{{ influxdb_meta_http_bind_address_port }}"
+    delay: 2
+    timeout: 3
+  when: influxdb_meta_started.changed and influxdb_meta_start_service
+  ignore_errors: yes
+  register: influxdb_meta_start_attempt
+
+- name: Assert status of influxdb-meta service
+  assert:
+    that:
+      - "influxdb_start_attempt == 0"
+  when:
+    - influxdb_meta_started.changed
+    - influxdb_meta_start_service
+    - influxdb_meta_start_attempt.failed is defined
+    - influxdb_meta_start_attempt.failed == True
+
+- name: Add meta nodes to cluster (run from first meta node)
+  command: "influxd-ctl add-meta {{ item }}:{{influxdb_meta_http_bind_address_port }}"
+  with_items: "{{ influxdb_meta_cluster_members }}"
+  delegate_to: "{{ influxdb_meta_cluster_leader }}"
+  when:
+    - influxdb_meta_cluster_auto_join == True
+    - influxdb_meta_cluster_leader == ansible_hostname
```

Comment:
I think this should be changed to use the first member in the group, since the leader may change with every run. For example, something more like:

```
- inventory_hostname == groups["meta-nodes"][0]
```